crates/ironrdp-pdu/src/rdp/suppress_output.rs

```diff
@@ -0,0 +1,57 @@
+use std::io;
+
+use byteorder::{ReadBytesExt as _, WriteBytesExt as _};
+use num_derive::{FromPrimitive, ToPrimitive};
+use num_traits::{FromPrimitive, ToPrimitive};
+
+use crate::{geometry::InclusiveRectangle, PduParsing};
+
+#[repr(u8)]
+#[derive(Debug, PartialEq, Eq, FromPrimitive, ToPrimitive, Clone)]
+pub enum AllowDisplayUpdatesType {
+    SuppressDisplayUpdates = 0x00,
+    AllowDisplayUpdates = 0x01,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct SuppressOutputPdu {
+    allow_display_updates: AllowDisplayUpdatesType,
+    desktop_rect: Option<InclusiveRectangle>,
```

Comment:
Thought: `allow_display_updates` seems redundant with `desktop_rect`.
> desktopRect (8 bytes): An Inclusive Rectangle (section 2.2.11.1) which contains the coordinates of the desktop rectangle if the allowDisplayUpdates field is set to ALLOW_DISPLAY_UPDATES (1). If the allowDisplayUpdates field is set to SUPPRESS_DISPLAY_UPDATES (0), this field MUST NOT be included in the PDU.

Suggestion: only keep `desktop_rect`, and infer `allow_display_updates` depending on whether it is `None` or `Some`.